content/docs/learning-guide/tutorials/rendering/vegetation-bending-tutorial.md

```diff
@@ -0,0 +1,704 @@
+
+## Prepare to add vegetation bending
+Now that we've edited our vertex shader, let's take the next step and prepare to add vegetation bending by adding an appropriate model.
+
+### Add a tree
+In order to test the code that we are about to write, we need an appropriate mesh!
+
+1. Open the **Editor** to your project and level. 
+1. In the **Entity Outliner**, **right-click** and select **Create entity**. **Right-click** the new entity in the **Entity Outliner**, and rename the entity to `Tree`.
+1. In the **Entity Inspector**, click **Add Component** and select **Mesh**.
+1. In the mesh component, under **Model Asset**, type `tree.fbx` or click the file button to find `tree.fbx` in your project's `Objects` folder.
+1. Again in the mesh component, click **Add Material Component**, click the file icon next to *Default Material* and select your material that you made earlier (`my_material`).
+
+Now, we have a tree (at an offset)! This tree is important because it uses vertex colors that we will be using to determine how the tree should bend.
+
+Note on vertex colors: You can use a DCC tool to color the vertices to match the type of bending you want to do on each part of the tree: 
+* Red vertex color: smaller movement with high frequency of random sinusoidal noise.
+* Green vertex color: delays the start of the movement for variations with high frequency of random sinusoidal noise.
+* Blue vertex color: larger movement and bending with low frequency of sinusoidal noise. 
+In the case of our tree, the trunk's vertices are colored blue and leaves are colored red.
+
+### Add a shader option
+Our tree mesh that we just added has colored vertices; however, not every mesh that you want to apply your material type to may have colored vertices. Therefore, we want to add a [*shader option*](/docs/atom-guide/dev-guide/shaders/azsl/#shader-variant-options) that allows the vertex shader to handle both of these cases. We also want to add the reference to the colors so we can use them in the code.
+
+1. Open `{your-project-path}\Materials\Types\VegetationBending_Common.azsli`.
+1. At the bottom of the file, add:
+
+   ```
+   option bool o_color_isBound;
+   ```
+   We place this variable in the common file because we will be using it in all the passes.
+1. Open `{your-project-path}\Materials\Types\VegetationBending_ForwardPass.azsli`. Inside `struct VegetationVSInput`, add another field:
+
+   ```
+   float4 m_optional_color : COLOR0;
+   ```
+   If the material's vertices are colored, `m_optional_color` will be set at runtime if it's available. Then, if `m_optional_color` is available, a soft naming convention will set `o_color_isBound` to true, which we can use later to determine if we want to perform the bending or not. Note that this soft naming convention is a very specific sub-feature of shader options that are set based on the presence or absence of an optional vertex stream, as opposed to shader options set based on material properties.
```

Comment:
Some revisions. Also, consider distinguishing this set of conceptual information.

Also, I assume you meant "mesh's vertices", not "material's vertices"?
```suggestion
   For a mesh with colored vertices, `m_optional_color` gets set at runtime, if it's available. Then, a soft name convention sets `o_color_isBound` to `true`, which we can use to determine if we want to perform the bending or not. 

   *Soft naming convention* is a very specific sub-feature of a shader option. It's set based on the presence or absence of an optional 
```